lib/app_profiler/middleware/upload_action.rb

```diff
@@ -4,9 +4,14 @@ module AppProfiler
   class Middleware
     class UploadAction < BaseAction
       class << self
-        def call(profile, response: nil, autoredirect: nil)
-          profile_upload = profile.upload
+        def call(profile, response: nil, autoredirect: nil, async: false)
+          if async
+            enqueue_upload(profile)
+            response[1][AppProfiler.profile_async_header] = true
```

Comment:
response[1] is a rack convention right? We don't need to index bound check it? What about nil check?